ttnn/cpp/ttnn/deprecated/tt_dnn/kernels/dataflow/generate_reduce_scaler.hpp

```diff
@@ -25,6 +25,7 @@ FORCE_INLINE void generate_reduce_scaler(const uint32_t cb_id, const uint32_t sc
     }
     noc_async_read_barrier();
 
+#ifndef REDUCE_ROW_SUM_VIA_MM
```

Comment:
Is the difference here that:
- regular scalar tile is first row of each face
- in reduce with mm, it's the first column of the tile

If this is the case, I think it would be better to split this function. `generate_reduce_scaler` is used as a helper in some other ops, so I think it would be cleaner if what it does is not affected by some compile-time define.

Reply:
Yes that is the case regular case has first row set to scalar in row-mayor tensor since we have tilized tensor there that maps to first row of each face to be se to scalar.
In reduce with mm we want first column to be set to scalar in row-mayor tensor, and in tilized version that means we have first column of faces 0 and 2 set to scalar and faces 1 and 3 set to zero.

I think this code should stay here because other ops that try to "fuse" reduce want to do the same thing and use reduce with mm, @johanna-rock-tt experimented with integrating this into distributed layer norm and scaled dot product attention.

Reply:
I think it would be clearer to have a `generate_reduce_with_mm_scaler` and have the `REDUCE_ROW_SUM_VIA_MM` define decide which helper to call inside the OP. The reason is I don't think we should encourage defines that implicitly control how code that is shared between OPs work. Each OP sees it's own datamovement and compute kernels and if defines are localized to that OP, it's easier to understand what is happening in the OP. The alternative is to properly document these shared defines `REDUCE_ROW_SUM_VIA_MM`. Then, I think it would be okay to have it inside the function.